tests/pages/courses/id.test.js

```diff
@@ -0,0 +1,169 @@
+        {
+          cohort_id: 'RU5mLpQrZZWlmftNSB2w', // Simulate the user being registered in a cohort
+          course_id: 'Rust_State_Machine',
+        },
+      ],
+    }
+
+    getUserFromFirestore.mockResolvedValue(subscription) // Mock updated user data retrieval
+  }
+
+  it('shows NotFound component if course is not active', async () => {
+    renderCourse({ course: { ...mockCourse, active: false } }) // Render with an inactive course
+    await waitFor(() => {
+      expect(screen.getByText('Oops!')).toBeInTheDocument() // Expect the "Oops!" message to be shown
+    })
+  })
+
+  it('renders the course title and description', async () => {
+    renderCourse() // Render the course component
+    expect(screen.getByText('Test Course')).toBeInTheDocument() // Expect course title to be displayed
+    expect(screen.getByText('A test course description')).toBeInTheDocument() // Expect course description to be displayed
+  })
+
+  it('allows user to register for the course', async () => {
+    renderCourse() // Render the course component
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+    expect(registerUserInCohortInFirestore).toHaveBeenCalledWith('RU5mLpQrZZWlmftNSB2w', 'user-123') // Ensure user registration is triggered
+  })
+
+  it('shows Calendar buttons after subscribing to the course', async () => {
+    renderCourse() // Render the course component
+
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+
+    await waitFor(async () => {
+      simulateUserRegistration() // simulate user registration in the build
+
+      const calendar = await screen.findByText('addToCalendar') // Ensure "Add to Calendar" button is present
+      const addToGoogleCalendar = await screen.findByText('addToGoogleCalendar') // Ensure "Add to Google Calendar" button is present
+
+      expect(calendar).toBeInTheDocument() // Check that calendar button is visible
+      expect(addToGoogleCalendar).toBeInTheDocument() // Check that Google Calendar button is visible
+    })
+  })
+
+  it('shows Discord and Wallet buttons after subscribing to the course', async () => {
+    renderCourse() // Render the course component
+
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+
+    await waitFor(async () => {
+      simulateUserRegistration() // simulate user registration in the build
```

Comment:
The original function name, `registeredUser`, suggests a static entity rather than an action, which can be confusing in the context of a callable function. A function name should clearly describe the action it performs.

I renamed it to `simulateUserRegistration()` to better reflect its purpose, indicating that the function is simulating the process of registering a user.